applications/luci-app-olsr-viz/htdocs/luci-static/resources/view/olsr-viz/olsr-viz-view.js

```diff
@@ -0,0 +1,178 @@
+				viz_res = result.viz_data;
+
+				var cssScript = E('script', {
+					'language': 'JavaScript',
+					'type': 'text/javascript'
+				}, ["var css=document.styleSheets[0];\n" +
+					"if (null != css.insertRule) {\n" +
+					"  css.insertRule('.label {color:black;background-color:white}', css.cssRules.length);\n" +
+					"}\n" +
+					"else {\n" +
+					"  css.addRule('.label', 'color:black');\n" +
+					"  css.addRule('.label', 'background-color:white');\n" +
+					"}"]);
+
+				var vizScript = E('script', {
+					'language': 'JavaScript1.2',
+					'src': L.resource(['olsr-viz.js']),
+					'type': 'text/javascript'
+				});
+
+				var mainDiv = E('div', {
+					'id': 'main',
+					'style': 'width: 100%; height: 93%; border: 1px solid #ccc; margin-left:auto; margin-right:auto; text-align:center; overflow: scroll'
+				}, [
+					E('div', { 'id': 'edges', 'style': 'width: 1px; height: 1px; position: relative; z-index:2' }),
+					E('div', { 'id': 'nodes', 'style': 'width: 1px; height: 1px; position: relative; z-index:4' })
+				]);
+
+				var zoomInput = E('input', {
+					'id': 'zoom',
+					'name': 'zoom',
+					'type': 'text',
+					'value': '2.0',
+					'size': '5',
+					'style': 'min-width: unset !important;',
+					'onchange': 'set_scale()'
+				});
+				var metricInput = E('input', {
+					'id': 'maxmetric',
+					'name': 'maxmetric',
+					'type': 'text',
+					'value': '3',
+					'size': '4',
+					'style': 'min-width: unset !important;',
+					'onchange': 'set_maxmetric(this.value)'
+				});
+				var autoOptimizationCheckbox = E('input', {
+					'id': 'auto_declump',
+					'name': 'auto_declump',
+					'type': 'checkbox',
+					'onchange': 'set_autodeclump(this.checked)',
+					'checked': 'checked'
+				});
+				var hostnamesCheckbox = E('input', {
+					'id': 'show_hostnames',
+					'name': 'show_hostnames',
+					'type': 'checkbox',
+					'onchange': 'set_showdesc(this.checked)',
+					'checked': 'checked'
+				});
+
+				var form = E('form', { 'action': '' }, [
+					E('p', {}, [
+						E('b', { 'title': 'Bestimmt die Vergrößerungsstufe.' }, 'Zoom '),
+						E('a', { 'href': 'javascript:set_scale(scale+0.1)' }, '+ '),
+						E('a', { 'href': 'javascript:set_scale(scale-0.1)' }, '\u2212 '),
+						zoomInput,
+						E('b', { 'title': 'Beschränkt die Anzeige auf eine maximale Hop-Entfernung.' }, '&nbsp;&nbsp;Metrik'),
+						E('a', { 'href': 'javascript:set_maxmetric(maxmetric+1)' }, '+ '),
+						E('a', { 'href': 'javascript:if(0<maxmetric)set_maxmetric(maxmetric-1)' }, '\u2212'),
+						metricInput,
+						E('b', { 'title': 'Schaltet die automatischen Layout-Optimierung ein.' }, '&nbsp;&nbsp;Optimierung'),
+						autoOptimizationCheckbox,
+						E('b', { 'title': 'Zeige Hostnamen an.' }, ' |  Hostnamen'),
+						hostnamesCheckbox,
+						E('a', { 'href': 'javascript:viz_save()', 'title': 'Speichert die aktuellen Einstellungen in einem Cookie.', 'style': 'font-weight:700;' }, '&nbsp;|&nbsp;&nbsp;Speichern'),
```

Comment:
I suggest to replace all `href: 'javascript:...'` with `href: '#', click: ...`, e.g. `E('a', { 'href': '#', 'click': viz_save, 'title': ...`
(This requires loading `olsr-viz.js` from `load()`

Reply:
I fixed all issues and tested it on my router as well. You can take a look. Thanks
https://github.com/openwrt/luci/compare/3b0a64c10c525d3ff4da2b89c8c05462faf9d549..f0ca73364f747064142ea31a46ea3173b4edef6b

Reply:
I also fixed the script.